src/mlpack/methods/ann/rbm.hpp

```diff
@@ -0,0 +1,183 @@
+ */
+#ifndef MLPACK_METHODS_ANN_RBM_HPP
+#define MLPACK_METHODS_ANN_RBM_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+#include <mlpack/methods/ann/activation_functions/softplus_function.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial neural networks.  */ {
+
+template<typename InitializationRuleType, typename RBMPolicy>
+class RBM
+{
+ public:
+  using NetworkType = RBM<InitializationRuleType, RBMPolicy>;
+  typedef typename RBMPolicy::ElemType eT;
+
+  /* 
+   * Intalise all the parameters of the network
+   * using the intialise rule. 
+   *
+   * @tparam IntialiserType rule to intialise the parameters of the network
+   * @param predictors training data
+   * @param numSteps Number of gibbs steps sampling
+   * @param useMonitoringCost evaluation function to use
+   * @param persistence indicates to use persistent CD
+   */
+  RBM(arma::Mat<eT> predictors, InitializationRuleType initializeRule,
+      RBMPolicy rbmPolicy,
+      const size_t numSteps = 1,
+      const size_t mSteps = 1,
+      const bool useMonitoringCost = true,
+      const bool persistence = false);
+
+  // Reset the network
+  void Reset();
+
+  /* 
+   * Train the network using the Opitimzer with given set of args.
+   * the optimiser sets the parameters of the network for providing
+   * most likely parameters given the inputs
+   * @param: predictors data points
+   * @param: optimizer Optimizer type
+   */
+  template<typename OptimizerType>
+  void Train(const arma::Mat<eT>& predictors, OptimizerType& optimizer);
+
+ /**
+  * Evaluate the rbm network with the given parameters.
+  * The function is needed for monitoring the progress of the network.
+  *
+  * @param parameters Matrix model parameters.
+  * @param i Index of point to use for objective function evaluation.
+  */
+  double Evaluate(const arma::Mat<eT>& parameters, const size_t i);
+
+ /** 
+  * This function calculates
+  * the free energy of the model
+  * @param: input data point 
+  */
+  double FreeEnergy(arma::Mat<eT>&& input);
+
+ /*
+  * This functions samples the hidden
+  * layer given the visible layer
+  *
+  * @param input visible layer input
+  * @param output the sampled hidden layer
+  */
+  void SampleHidden(arma::Mat<eT>&& input, arma::Mat<eT>&& output);
+
+  /*
+  * This functions samples the visible
+  * layer given the hidden layer
+  *
+  * @param input hidden layer
+  * @param output the sampled visible layer 
+  */
+  void SampleVisible(arma::Mat<eT>&& input, arma::Mat<eT>&& output);
+
+ /*
+  * This function does the k-step
+  * gibbs sampling.
+  *
+  * @param input: input to the gibbs function
+  * @param output: stores the negative sample
+  * @param steps: number of gibbs sampling steps
+  */
+  void Gibbs(arma::Mat<eT>&& input, arma::Mat<eT>&& output,
```

Comment:
Can you align the parameter with the rest and use a new line for the output parameter, if the line ends up being longer than 80 characters.